services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/extension/fixture/rule/CleanupSwitchesRule.groovy

```diff
@@ -25,24 +27,66 @@ class CleanupSwitchesRule extends AbstractRuleExtension<CleanupSwitches> {
 
     @Override
     void afterTest() {
-        log.debug "Verifying that all the switches are clean"
-        try {
-            Wrappers.wait(WAIT_OFFSET) {
-                topology.activeSwitches.each { sw ->
+        topology.activeSwitches.each { sw ->
+            try {
+                Wrappers.wait(WAIT_OFFSET) {
                     def rules = northbound.validateSwitchRules(sw.dpId)
-                    assert rules.missingRules.empty
-                    assert rules.excessRules.empty
-                    assert rules.properRules.empty
+                    rules.excessRules.empty ?: failWith(new ExcessRulesException(
+                            "$sw.dpId switch appear to have excess rules at the end of the test.", rules))
```

Comment:
I think this variant is more cute.

"Switch '$sw.dpId' appears to have excess rules at the end of the test. "

Please, note
1. appear -> appears
2. The sentence is followed by a whitespace

Reply:
why to follow with a whitespace?

Reply:
Yes, you are right. But in this exception, I would like that rules will not be printed on a new line. Though it is not a big deal. Probably printing rules on a new line would be even better.

Reply:
did similar to WaitException